### PR TITLE
Allow sql_package: "go-sql-driver/mysql"

### PR DESCRIPTION
**Pull request by Jille**
In preparation of #2179 for LOAD DATA LOCAL INFILE

### CODE REVIEW

File: internal/codegen/golang/driver.go [hunk 1]
```diff
@@ -35,7 +39,18 @@ func (d SQLDriver) Package() string {
...
 }
+
+func (d SQLDriver) UsesStandard() bool {
+	switch d {
+	case SQLDriverPGXV4, SQLDriverPGXV5:
+		return false
+	default:
+		return true
+	}
+}
```
Comment (kyleconroy) [inline internal/codegen/golang/driver.go:56]: Can we move this to another PR? This appears to be unused in this PR.

File: internal/codegen/golang/driver.go [hunk 2]
```diff
@@ -3,14 +3,16 @@ package golang
 type SQLDriver int
 
 const (
-	SQLPackagePGXV4    string = "pgx/v4"
-	SQLPackagePGXV5    string = "pgx/v5"
-	SQLPackageStandard string = "database/sql"
+	SQLPackagePGXV4            string = "pgx/v4"
+	SQLPackagePGXV5            string = "pgx/v5"
```
Comment (kyleconroy) [inline internal/codegen/golang/driver.go:7]: I'm starting to think we should have called this `jackc/pgx/v5` or `github.com/jackc/pgx/v5`. Happy to keep `go-sql-driver/mysql` for now.

**Review by kyleconroy — CHANGES_REQUESTED** [review]
I think we want to add a new configuration option for this instead of re-using `sql_package`. The current value changes the package to use, either database/sql or pgx/v4 or pgxv5. In this case we're still using database/sql as a base, but using the mysql driver to do some additional work.

I think adding a new `sql_driver` override is the way to go and require it to be set to the full import path of `github.com/go-sql-driver/mysql`.

File: internal/codegen/golang/driver.go (hunk 2 above)
Comment (Jille) [reply internal/codegen/golang/driver.go]: Easy enough to start accepting the new values as well as the old values at some point, right?

File: internal/codegen/golang/driver.go (hunk 1 above)
Comment (Jille) [reply internal/codegen/golang/driver.go]: Good call, I didn't even need it in the next PR. Removed.

**Comment by Jille** [comment]
Done

**Review by kyleconroy — APPROVED** [review]
